include/ssm.h

```diff
@@ -178,10 +187,7 @@ struct ssm_mm {
  *
  *  @param v  pointer to the #ssm_mm header of the heap item.
  */
-#define ssm_dup_unsafe(v)                                                      \
-  do                                                                           \
-    ++(v).heap_ptr->ref_count;                                                 \
-  while (0)
+#define ssm_dup_unsafe(v) ((++(v).heap_ptr->ref_count, (v)))
```

Comment:
I adjusted this macro to be an expression that returns the given value, so we can use it as an expression in other macros like `some_function(ssm_dup(v))`.

However, we need to be very careful about using this in macros where the expression may be duplicated, which will lead to leaked memory.